ex_39_v1.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+# employee_names = ['john johnson', 'tou xiong', 'michaela michaelson', 'jake jacobson', 'jacquelyn jackson', 'sally weber']
+# sorted_employee_names = sorted(employee_names)
+
+
+# new_list_sorted_employee_names = []
+    
+# for name in sorted_employee_names:
+#     new_list_sorted_employee_names.append(str(name))
+
+employee_1 = {
+    'first_name': 'tou',
+    'last_name': 'xiong',
+    'position': 'software engineer',
+    'separation_date': '2016-10-05',
+}
+
+employee_2 = {
+    'first_name': 'john',
+    'last_name': 'johnson',
+    'position': 'manager',
+    'separation_date': '2016-12-31',
+}
+
+employee_3 = {
+    'first_name': 'michaela',
+    'last_name': 'michaelson',
+    'position': 'district manager',
+    'separation_date': '2015-12-19',
+ }
+
+employee_4 = {
+    'first_name': 'jake',
+    'last_name': 'jacobson',
+    'position': 'programmer',
+    'separation_date': ' ',
+ }
+
+employee_5 = {
+    'first_name': 'jacquelyn',
+    'last_name': 'jackson',
+    'position': 'dba',
+    'separation_date': ' ',
+ }
+
+employee_6 = {
+    'first_name': 'sally',
+    'last_name': 'weber',
+    'position': 'web developer',
+    'separation_date': '2015-12-18',
+ }
+
+employees = [employee_1, employee_2, employee_3, employee_4, employee_5, employee_6]
+
+for employee in employees:
```

Comment:
Okay, so the order that these are printed in is based on their order in the `employees` list. First, try generating a new list that is comprised of the six last name's in alphabetical order. So it should look like this when you're done:

```
['jackson', 'jacobson', 'johnson', 'michaelson', 'weber', 'xiong']
```

Reply:
We are solving the easier problem first...just get the last names in alphabetical order, and then we'll use this as the eventual loop to solve the harder problem: looping through the dictionaries in alphabetical order.

Reply:
last_names_in_alpha_order = ['johnson', 'xiong', 'michaelson', 'jacobson', 'jackson', 'weber']
last_names_in_alpha_order.sort()

print(last_names_in_alpha_order)

Reply:
Okay, so now that you have this, you need to iterate through this list instead of `for employee in employees:`

Each iteration will give you the last name...then you need to look up that person in the `employees` list (containing the dictionaries of each person) and then the rest of your code should work. Does that make sense?

Recall that you can look up someone in the `employees` list by iterating through the list and testing to see if the `last_name` value matches the last name that you're looking for.